src/main/java/org/gridsuite/study/server/StudyService.java

```diff
@@ -96,22 +99,39 @@ public StudyService(
         this.webClient =  webClientBuilder.build();
 
         this.studyRepository = studyRepository;
+        this.studyBySubjectRepository = studyBySubjectRepository;
+        this.studyByAccessRightsRepository = studyByAccessRightsRepository;
     }
 
-    Flux<StudyInfos> getStudyList() {
-        Flux<Study> studyList = studyRepository.findAll();
-        return studyList.map(study -> new StudyInfos(study.getName(), study.getDescription(), study.getCaseFormat()));
+    Flux<StudyInfos> getStudyList(String subject) {
+        Flux<StudyBySubject> studyBySubjectFlux = studyBySubjectRepository.findAllBySubject(subject);
+
+        return Flux.concat(studyBySubjectFlux.flatMap(studyBySubject -> {
+            Mono<Study> studyMono = studyRepository.findByName(studyBySubject.getStudyName());
```

Comment:
I think we should denormalize the data to avoid having to do 1 request per study ?